### PR TITLE
Track power energy

### PR DESCRIPTION
**Pull request by patrik3k**
-Added a monthly energy meter for current month.  (This code is not optimal but as we dont have any readings in the future so its safe to assume we can read measures for 31 days.)
-Changed current power sensor so it uses "latest" in API call instead of time period as for the power is just the current reading that is relevant. 
-Changed the ..._power_2 entity name to ..._energy for the energy sensor
-Added NGenic prefix to all names so its easier to find and maintain sensors in HA when having many sensors and devices. 

Sorry if the code is not professional level.

### CODE REVIEW

File: sensor.py [hunk 1]
```diff
@@ -26,6 +26,27 @@
 
 TIME_ZONE = "Z" if str(dt_util.DEFAULT_TIME_ZONE) == "UTC" else str(dt_util.DEFAULT_TIME_ZONE)
 
+def get_from_to_datetime_month(months=1):
```
Comment (sfalkman) [inline sensor.py]: `months` are never used

File: sensor.py [hunk 2]
```diff
@@ -26,6 +26,27 @@
...
+
+    Both dates include the time zone name, or `Z` in case of UTC.
+    Including these will allow the API to handle DST correctly. 
+
+    When asking for measurements, the `from` datetime is inclusive
+    and the `to` datetime is exclusive. 
+    """
+    #to_dt = datetime.datetime.now()
+    from_dt = datetime.datetime.now().replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+    to_dt = from_dt + datetime.timedelta(days=31)
```
Comment (sfalkman) [inline sensor.py]: This will probably not work as expected, as it will spill over in the next month if it is less than 31 days.

File: sensor.py [hunk 3]
```diff
@@ -188,6 +220,18 @@ def unit_of_measurement(self):
         """Return the unit of measurement."""
         return "kW"
 
+
```
Comment (sfalkman) [inline sensor.py]: Remove these empty lines

File: sensor.py (hunk 2 above)
Comment (sfalkman) [reply sensor.py]: I just re-read the pull request comment about this, can you please add a comment that it's a non-issue to use 31 days if you're keeping it?

**Comment by sfalkman** [comment]
I left a couple of comments, otherwise I think it looks good.
Using the `latest` API for the power sensor must be something new in the API as it didn't work before, good update! (I see that Ngenic is on v3 of the API, last time I made changes they were on v1).

@robinostlund if you got the time, it would be great if you could test patrik3k's fork before merging as we went through a couple of quirks in the initial track implementation.

File: Contributors.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Contributors
```
Comment (sfalkman) [inline Contributors.md]: Can you remove this, I think it shows automatically in github which are the contributors, and I don't want to maintain a list :)

File: sensor.py (hunk 2 above)
Comment (patrik3k) [reply sensor.py]: Added comment, and fixed  so its properly is just the month we return from 1st to 1st.

File: sensor.py (hunk 3 above)
Comment (patrik3k) [reply sensor.py]: Fixed

File: Contributors.md (hunk 4 above)
Comment (patrik3k) [reply Contributors.md]: Ok, removed, I just followed the newbie instructions you sent. :)

**Comment by patrik3k** [comment]
Do I need to create a new pull request now?

File: sensor.py (hunk 1 above)
Comment (patrik3k) [reply sensor.py]: Removed.

**Comment by sfalkman** [comment]
Looks good. One last question: if you compare the power/energy from homeassistant and the ngenic app, do they differ? If you look at both monthly and current usage?

**Comment by patrik3k** [comment]
Yes the numbers add up except that its a rounding error that looses precision in the current power, daily and monthly is correct. The current power returns a float in the API but is rounded to 1 decimal in the sensor. This is a loss of information and the App showed 1.8kW but the sensor showed 1.7 and the result from API was 1.799580097198486 . This is rounded to 1.8 in python round(x,1 ) but there might have been a timing issue. The next reading was 1.8396525382995605 and the sensor showed 1.8.  Most power sensors show W and and not kW I suggest we change to W and increas the presion to 5 digits ie. 1839.4 W instedad of 1.8kW. This is a breaking change for anyone that uses  the data in an automation. I'll update the sensor to 4 digits precision and kW for now, let me know if you think we shoud report Ws instead?

**Comment by patrik3k** [comment]
After further investigation the correct way to report power is in Watts as in homeassistant.const
only the POWER_WATT constant is defined. I changed the code to return watts instead by just multiplying the value with 1000 and rounded to one decimal.

**Comment by patrik3k** [comment]
> Do you really want to display the watt? You're measuring the whole house so it would likely be in the kilo-area, no?

Yes, I have about 15 power sensors in my house. Many are in the low power range 0-100W but a few ranges from 0 to a few thousand Watts. They are all measured in Watts and you can easily put them together in one graph or sum them together in a template. The standard for home assistant is watts. If someone want to present it differently its easy to create a template for it. I'll say go with watts but you are the boss.

**Comment by sfalkman** [comment]
Yeah you're right, it's probably better to have it in W since other power sensors use that as well.

All of this looks good to me, do you have some further changes or should I merge?

**Comment by patrik3k** [comment]
Its all good, please merge. 
I tried to make different update frequencies for the different sensors as the  power is continuously updated but the energy is updated less frequent and the monthly would be enough on an hourly basis. I failed after adding different SCAN_INTERVALLS but it didnt work so I threw it away. If you have knowledge on how to do it its greatly appreciated. Its not good if the component hits a throttle limit now when it makes more calls to the API.

**Comment by sfalkman** [comment]
Sounds like a good idea, I did some tests and got separate intervals to work but let's fix that outside of this PR.

Good work, thanks for the contribution! :)

**Pull request merged**